### PR TITLE
Typing for schema registry serializer deserializer

### PR DESCRIPTION
**Pull request by hackaugusto**
~I'm going through the code base and adding some type annotations, opened this PR because it *may* have found a bug.~ There was no bug in the code, but in the typing.

I'm still keeping the PR open while the changes are small :)

### CODE REVIEW

File: karapace/serialization.py [hunk 1]
```diff
@@ -108,63 +109,64 @@ async def close(self):
...
 
-    async def close(self):
+    async def close(self) -> None:
         if self.registry_client:
             await self.registry_client.close()
             self.registry_client = None
 
     def get_subject_name(self, topic_name: str, schema: str, subject_type: str, schema_type: SchemaType) -> str:
-        schema = TypedSchema.parse(schema_type, schema)
+        schema_typed = TypedSchema.parse(schema_type, schema)
```
Comment (hackaugusto) [inline karapace/serialization.py:128]: I introduced new names because the types differ, and type checking fails.

File: karapace/serialization.py [hunk 2]
```diff
@@ -108,63 +109,69 @@ async def close(self):
...
         if schema_type is SchemaType.AVRO:
-            namespace = schema.schema.namespace
+            namespace = schema_typed.schema.namespace
         if schema_type is SchemaType.JSONSCHEMA:
-            namespace = schema.to_json().get("namespace", "dummy")
+            namespace = schema_typed.to_json().get("namespace", "dummy")
         return f"{self.subject_name_strategy(topic_name, namespace)}-{subject_type}"
 
     async def get_schema_for_subject(self, subject: str) -> TypedSchema:
+        assert self.registry_client, "must not after the object is closed."
```
Comment (gabriel-tincu) [inline karapace/serialization.py:142]: must not what?

File: karapace/serialization.py [hunk 3]
```diff
@@ -108,63 +109,69 @@ async def close(self):
...
+        schema_id = await self.registry_client.post_new_schema(subject, schema_typed)
         async with self.state_lock:
-            self.schemas_to_ids[ser_schema] = schema_id
-            self.ids_to_schemas[schema_id] = schema
+            self.schemas_to_ids[schema_ser] = schema_id
+            self.ids_to_schemas[schema_id] = schema_typed
         return schema_id
 
     async def get_schema_for_id(self, schema_id: int) -> TypedSchema:
+        assert self.registry_client, "must not after the object is closed."
```
Comment (gabriel-tincu) [inline karapace/serialization.py:166]: same as above

**Review by gabriel-tincu — DISMISSED** [review]
Other than the text for the 2 assertions looking a bit confusing, it LGTM. Either I'm missing something there or it needs updating

File: karapace/serialization.py (hunk 2 above)
Comment (hackaugusto) [reply karapace/serialization.py]: Ooops, my bad, updated it to `must not call this method after the object is closed.`

**Review by gabriel-tincu — APPROVED** [review]

**Pull request merged**